# Generated by Django 4.2.6 on 2023-12-02 12:41

import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('recipes', '0007_alter_recipeingredient_ingredient_and_more'),
    ]

    operations = [
        migrations.RemoveConstraint(
            model_name='favorites',
            name='unique_favorites_recipe_user',
        ),
        migrations.RemoveConstraint(
            model_name='ingredient',
            name='unique_name_measurement_unit',
        ),
        migrations.RemoveConstraint(
            model_name='recipe',
            name='unique_name_author',
        ),
        migrations.RemoveConstraint(
            model_name='shoppingcart',
            name='unique_shoppingcard_recipe_user',
        ),
        migrations.RemoveConstraint(
            model_name='tag',
            name='unique_name_color_slug',
        ),
        migrations.AlterField(
            model_name='tag',
            name='color',
            field=models.CharField(max_length=7, validators=[django.core.validators.RegexValidator(message='Цвет не соответствует формату HEX.', regex='^\\#([a-fA-F0-9]{6})$')], verbose_name='Цвет тега'),
        ),
        migrations.AlterField(
            model_name='tag',
            name='slug',
            field=models.SlugField(error_messages={'unique': 'Вводимый slug уже имеется.'}, max_length=200, unique=True, verbose_name='Слаг тега'),
        ),
        migrations.AddConstraint(
            model_name='favorites',
            constraint=models.UniqueConstraint(fields=('recipe', 'user'), name='unique_favorites_recipe_user', violation_error_message='Рецепт уже в Избранном'),
        ),
        migrations.AddConstraint(
            model_name='ingredient',
            constraint=models.UniqueConstraint(fields=('name', 'measurement_unit'), name='unique_name_measurement_unit', violation_error_message='Ингредиент уже имеется в БД, если вы хотите изменить еденицу измерения, необходимо внести корректировки в файл CSV.'),
        ),
        migrations.AddConstraint(
            model_name='recipe',
            constraint=models.UniqueConstraint(fields=('name', 'author'), name='unique_name_author', violation_error_message='Вы уже создавали рецепт с таким названием'),
        ),
        migrations.AddConstraint(
            model_name='shoppingcart',
            constraint=models.UniqueConstraint(fields=('recipe', 'user'), name='unique_shoppingcard_recipe_user', violation_error_message='Рецепт уже в Корзине'),
        ),
        migrations.AddConstraint(
            model_name='tag',
            constraint=models.UniqueConstraint(fields=('name', 'color', 'slug'), name='unique_name_color_slug', violation_error_message='Тег с выбранным цветом slug уже существует, введите другой тег с другим цветом и другим slug.'),
        ),
    ]
